source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Charges/Handlers/ChargeCommandReceivedEventHandler.cs

```
@@ -84,14 +80,13 @@ await _chargeCommandReceiptService
                     await HandleCreateEventAsync(commandReceivedEvent.Command).ConfigureAwait(false);
                     break;
                 case OperationType.Update:
-                    if (charge == null)
-                        throw new InvalidOperationException("Could not update charge. Charge not found.");
                     HandleUpdateEvent(charge, commandReceivedEvent.Command);
                     break;
                 case OperationType.Stop:
-                    if (charge == null)
-                        throw new InvalidOperationException("Could not stop charge. Charge not found.");
-                    charge.Stop(commandReceivedEvent.Command.ChargeOperation.EndDateTime);
+                    charge?.Stop(commandReceivedEvent.Command.ChargeOperation.EndDateTime);
```

Is it correct to silently ignore all operations if the charge is not found?

---

No we should not continue if null. I will change it to use the null-forgiving operator - as null will always be treated as a `OperationType.Create`

This is all subject to change, so we are aware of the temporary bad design.